### PR TITLE
utils prometheusmetrics: convert gauges to counters

### PR DESCRIPTION
**Pull request by dshehbaj**
**What type of PR is this?**

Bug Fix

**Which issue does this PR fix?**:

#3031 

**What does this PR do / Why do we need it?**:
Bug fix

**Testing done on this change**:

Added unit tests confirming the behavior of the counters.

**Will this PR introduce any new dependencies?**:

No

**Will this break upgrades or downgrades? Has updating a running cluster been tested?**:
No

**Does this change require updates to the CNI daemonset config files to work?**:

No

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/ipamd/ipamd.go [hunk 1]
```diff
@@ -2346,3 +2346,13 @@ func (c *IPAMContext) AddFeatureToCNINode(ctx context.Context, featureName rcv1a
 	newCNINode.Spec.Features = append(newCNINode.Spec.Features, newFeature)
 	return c.k8sClient.Patch(ctx, newCNINode, client.MergeFromWithOptions(cniNode, client.MergeFromWithOptimisticLock{}))
 }
+
+func (c *IPAMContext) tryAssignPodENI(ctx context.Context, pod *corev1.Pod, fnName string) error {
```
Comment (jaydeokar) [inline pkg/ipamd/ipamd.go:2350]: Why do we need this ?

Comment (dshehbaj) [reply pkg/ipamd/ipamd.go:2350]: Using this twice in the tests as a helper func:

https://github.com/dshehbaj/amazon-vpc-cni-k8s/blob/shehbaj/addipcnt-gauge-bug/pkg/ipamd/ipamd_test.go#L2437

https://github.com/dshehbaj/amazon-vpc-cni-k8s/blob/shehbaj/addipcnt-gauge-bug/pkg/ipamd/ipamd_test.go#L2447

I can move this helper func into the test module if we do not want to keep this in the main package.

Comment (dshehbaj) [reply pkg/ipamd/ipamd.go]: I moved this function to the test package now:

https://github.com/dshehbaj/amazon-vpc-cni-k8s/blob/shehbaj/addipcnt-gauge-bug/pkg/ipamd/ipamd_test.go#L2462

**Review by jaydeokar — APPROVED** [review]
lgtm

**Review by orsenthil — APPROVED** [review]
LGTM

**Pull request merged**